src/components/TrackSettingsButton.js

```diff
@@ -15,8 +19,17 @@ export const TrackSettingsButton = ({
     label,
     availableColors
 }) => {
+    // based off of https://react-popup.elazizi.com/controlled-popup/#using-open-prop
+    const [open, setOpen] = useState(false);
+    const close = () => setOpen(false);
     return(
-        <Popup trigger={<button type="button"> Settings </button>} position="right center" contentStyle={{width: "760px"}} modal>
+      <div>
```

Comment:
I think this `<div>` and `</div>` could be replaced with `<>` and `</>` and save an unnecessary HTML element, but it should work fine as is.